File: Array/reverseArray.c
#include<stdio.h>

int reverseArray(int arr[],int n)
{
 int start =0,end = n-1,temp;
 while(start<end)
 {
  temp = arr[start];
  arr[start] = arr[end];
  arr[end] = temp;
  start++;
  end--;
 }
 
}

int main()
{
 int arr[]= {8,5,12,1,7,15};
 int n = sizeof(arr)/sizeof(arr[0]);
 reverseArray(arr,n);
 for(int i=0;i<n;i++)
 {
  printf("%d\t",arr[i]);
 }
 printf("\n");
 return 0;
}

File: Array/reverseArray.py
'''
def reverseArray(arr,start,end):
	while(start<end):
		temp = arr[start]
		arr[start] = arr[end]
		arr[end] =temp
		start += 1
		end -= 1 
'''

def reverseArray(A,start,end):
	while start<end:
		A[start], A[end] = A[end], A[start]
		start += 1;
		end -= 1;



arr = [15,7,1,12,5,8]
print("Original array:",arr)

reverseArray(arr,0,len(arr)-1)
print("Reverse array:",arr)
